### DISCUSSION TITLE
Directory stored in pimple container

### DISCUSSION THREAD

**Discussion by ssnepenthe** [vimeo/psalm#6525] [Q&A] [answered]
Given the following:

```php
$container = new \Pimple\Container(['dir' => __DIR__]);

require_once $container['dir'] . '/some-file.php';
```

Psalm is reporting:

```
Cannot resolve the given expression to a file path
```

Is it possible for me to help Psalm understand `$container['dir']`?

**Comment by orklah** [comment, marked as answer]
I think this would require some plugin that would store the values from the constructor argument and provide them back on each access.

**Reply by ssnepenthe** [reply]
Good to know, thanks!
